src/test_workflow/smoke_test/smoke_test_cluster_opensearch.py

```diff
@@ -0,0 +1,99 @@
+# Copyright OpenSearch Contributors
```

Comment:
I believe we are duplicating the work of setting up clusters here. Why not just use the integ-test cluster set up instead of creating a new framework for smoke testing? Check the set ups https://github.com/opensearch-project/opensearch-build/tree/main/src/test_workflow/integ_test and see if you can use the same classes to set up a cluster for smoke testing as well. From high level I do not see a difference in the set up.

Reply:
Reusing the entire integ-test cluster might be overcomplicate for smoke test as we don't want to accept any customized configuration for smoke tests cluster to keep it lightweight. I also want to make smoke test workflow less dependent on integ tests as a standalone test workflow.

Reply:
Any example API that I can test for system templates plugin you are referring? Anyway I think if certain component requires specific configuration for smoke test, it's not ready to be onboarded to smoke tests framework. Because we only deploy the cluster once and run all API request checks against it. We are not sure whether that specific configuration would affect others. 
We are reusing the `distributions` classes from integ tests workflow to install and start the cluster. For any future distribution type, it can be easily adopted by this smoke test workflow.

Reply:
Agree with reusing the cluster setup from integ test, the code is pretty modular to be reused here. To be able to even run an api of system-templates you need those configurations added to opensearch.yml. 
Think about adding smoke tests for features that need to be enabled explicitly, for e.g. concurrent search. The current implementation is pretty robust and reusable  `

Reply:
I still think we shouldn't allow any customized configurations as it might grow much bigger. We provide generic framework with all default configurations for the cluster by design. If any of the component needs specific configuration to be operational, it shouldn't be added to the smoke tests and can go into the integ tests.

Reply:
We can start with this approach but overall I think we will need the additional config for plugins specifically. Another use-case is if a new plugin or feature is added that is experimental, we might need to enable it using additional-config. 
Again this is conditional in defining what qualifies as smoke testing.

Reply:
The requirements of smoke tests is set for default configuration and we don't want any additional configuration affect others test cases. Since smoke tests won't be the only exit criteria for RC, i think these tests cases can and shall be added to integ tests with more configuration options. Anyway, we may discuss more on that as an enhancement.